python/cudf/cudf/utils/ioutils.py

```diff
@@ -1596,29 +1585,21 @@ def generate_pandas_metadata(table: cudf.DataFrame, index: bool | None) -> str:
                     index_levels.append(materialized_idx)
                     columns_to_convert.append(materialized_idx._values)
                     col_names.append(descr)
-                    types.append(np_to_pa_dtype(materialized_idx.dtype))
+                    types.append(pa.from_numpy_dtype(materialized_idx.dtype))
             else:
                 descr = _index_level_name(
                     index_name=idx.name, level=level, column_names=col_names
                 )
                 columns_to_convert.append(idx._values)
                 col_names.append(descr)
-                if isinstance(idx.dtype, cudf.CategoricalDtype):
-                    raise ValueError(
```

Comment:
Have we changed the exception we're raising here? It looks like `cudf_dtype_to_pa_type` raises a NotImplementedError instead of a ValueError when the argument is a categorical. That's probably OK since this is already not an exception that we're trying to match to pandas, I'm just double-checking.